### PR TITLE
feat(service-portal): new components

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

Add a few components and a hook
* Timeline
* ProgressBar
* IsMobile hook

## Why

For intellectual properties

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by thordurhhh — COMMENTED** [review]
Just a little translation missing. Otherwise nice.

File: libs/service-portal/assets/src/components/Timeline/Timeline.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            position="relative"
+            background="blue100"
+            borderRadius="large"
+            width="full"
+          >
+            {currentProgress && (
+              <>
+                <ProgressBar progress={currentProgress} />
+                {tooltip && (
+                  <Tooltip text={tooltipText} placement="top">
```
Comment (thordurhhh) [inline libs/service-portal/assets/src/components/Timeline/Timeline.tsx:86]: Is there any case where someone would want a `tooltip` without a `tooltipText` ?
```suggestion
                {tooltipText && (
                  <Tooltip text={tooltipText} placement="top">
```

Maybe just depend on the text and remove the tooltip prop or vice versa

**Review by thordurhhh — APPROVED** [review]
LGTM. Just a little suggestion.

File: libs/service-portal/assets/src/components/Timeline/Timeline.tsx (hunk 1 above)
Comment (thorkellmani) [reply libs/service-portal/assets/src/components/Timeline/Timeline.tsx:86]: yeah ok 💅

**Review by Tryggvig — APPROVED** [review]

**Pull request merged**